examples/server/server.cpp

```diff
@@ -1291,22 +1347,30 @@ int main(int argc, char **argv)
         llama.beginCompletion();
 
         if (!llama.stream) {
-            size_t stop_pos = std::string::npos;
+            if (llama.params.n_beams) {
```

Comment:
Here is an example of where it is set and used:
https://github.com/ggerganov/llama.cpp/blob/d59bd97065cd7ded6c4ecab54b1d5e0b1b11e318/examples/beam-search/beam-search.cpp#L110

In `examples/server/server.cpp` I believe it may be set via the command line.

I don't think beam search and grammar will currently work together. That is currently an open item: https://github.com/ggerganov/llama.cpp/issues/2923